### PR TITLE
Test /wallet/update

### PR DESCRIPTION
**Pull request by Vavilen**
For [#642](https://github.com/skycoin/skycoin/issues/642)

### CODE REVIEW

File: src/gui/wallet.go [hunk 1]
```diff
@@ -269,7 +270,7 @@ func walletNewAddresses(gateway *daemon.Gateway) http.HandlerFunc {
 }
 
 // Update wallet label
-func walletUpdateHandler(gateway *daemon.Gateway) http.HandlerFunc {
+func walletUpdateHandler(gateway Gatewayer) http.HandlerFunc {
```
Comment (gz-c) [inline src/gui/wallet.go]: Can you make this method POST-only

and add documentation for the endpoint to gui/README.md

File: src/gui/wallet_test.go [hunk 2]
```diff
@@ -555,3 +560,121 @@ func TestWalletGet(t *testing.T) {
...
+			"/wallet/transactions",
+			&httpBody{
+				WalletID: "foo",
+				Label:    "label",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - update wallet label failed: gateway.UpdateWalletLabel error",
+			"foo",
+			"label",
+			errors.New("gateway.UpdateWalletLabel error"),
```
Comment (gz-c) [inline src/gui/wallet_test.go]: Same here, change endpoint to do 404 if ErrWalletNotExist otherwise 500

File: src/gui/wallet_test.go [hunk 3]
```diff
@@ -555,3 +560,121 @@ func TestWalletGet(t *testing.T) {
...
+			}
+			if tc.body.Label != "" {
+				v.Add("label", tc.body.Label)
+			}
+		}
+		if len(v) > 0 {
+			url += "?" + v.Encode()
+		}
+		req, err := http.NewRequest(tc.method, url, nil)
+		if err != nil {
```
Comment (gz-c) [inline src/gui/wallet_test.go]: require.NoError

**Comment by Vavilen** [comment]
@gz-c , done.

**Pull request merged**